Feature/docker xenial

tomereli opened the pull request:
To make sure old GCC versions are tested, add build-in-docker-xenial
step to .gitlab-ci.yml.

Commit:
tools: docker: builder: add xenial Dockerfile

Add reference Dockerfile for ubuntu:xenial. This will be used to build
prplMesh using older GCC5.4.0.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

arnout commented on tools/docker/builder/xenial/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+FROM ubuntu:xenial
```
In the other one we call it ubuntu:18.04, so
```suggestion
FROM ubuntu:16.04
```

arnout commented on tools/docker/builder/xenial/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+    libnl-genl-3-dev \
+    libreadline-dev \
+    libssl-dev \
+    libssl-dev \
```
Twice?
```suggestion
    libssl-dev \
```

arnout commented on .gitlab-ci.yml:
```diff
@@ -111,6 +110,12 @@ build-in-docker-alpine:
 
+build-in-docker-xenial:
+  extends: build-in-docker
+  image: prplfoundationinc/prplmesh-builder:xenial
```
Please build the image in CI as well, and use the new naming.

arnout commented on .gitlab-ci.yml:
```diff
@@ -111,6 +110,12 @@ build-in-docker-alpine:
+  extends: build-in-docker
+  image: prplfoundationinc/prplmesh-builder:xenial
+  # only:
+  #   - schedules
```
Please don't leave in this commented-out stuff.

arnout commented on .gitlab-ci.yml:
```diff
@@ -250,12 +255,12 @@ run-certification-tests:
+          echo "Deploying to $DEVICE_UNDER_TEST"
+          tools/deploy_ipk.sh --certification-mode $DEVICE_UNDER_TEST build/$DEVICE_UNDER_TEST/prplmesh.ipk
+      fi
+    - /easymesh_cert/run_test_file.sh -v -o logs -d $DEVICE_UNDER_TEST $TESTS_TO_RUN
```
Please do the formatting corrections in a separate commit. (We should probably run yamllint on all yaml files...)

Commit:
.gitlab-ci.yml: build using the ubuntu-16.04 image

To make sure old GCC versions are tested, add build-in-docker-xenial
step to .gitlab-ci.yml, which builds prplmesh dummy in Ubuntu 16.04 with
GCC 5.4.0.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
.gitlab-ci.yml: run yaml formatting

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
